### PR TITLE
Compile with Cabal-3.0

### PR DESCRIPTION
**Pull request by phadej**
Revision checking doesn't work correctly if there are public multi-lib dependencies.

### CODE REVIEW

File: hackage-cli.cabal [hunk 1]
```diff
@@ -89,17 +89,17 @@ executable hackage-cli
     , filepath              ^>= 1.4.0.0
     , http-io-streams       ^>= 0.1.0.0
     , io-streams            ^>= 1.5.0.1
-    , lens                  ^>= 4.17
+    , lens                  ^>= 4.17 || ^>=4.18
```
[the remark this answers is not in the document]
Comment (phadej) [reply hackage-cli.cabal]: I'll keep in mind that for the future

**Comment by bgamari** [comment]
What is preventing this from being merged?

**Comment by hvr** [comment]
> What is preventing this from being merged?

Mostly a reasonably careful code-review... ;-)

File: lib/Distribution/Server/Util/CabalRevisions.hs [hunk 2]
```diff
@@ -442,28 +454,30 @@ checkDependencies componentName ds1 ds2 = do
 
     depKind = depTypeName dproxy ++ " dependency"
 
-class (Ord (DepKey d)) => IsDependency d where
+class (Ord (DepKey d), Pretty vr, Eq vr) => IsDependency vr d | d -> vr where
```
Comment (bgamari) [inline lib/Distribution/Server/Util/CabalRevisions.hs]: Out of curiosity, why a functional dependency here where `DepKey` is an associated type?

**Comment by phadej** [comment]
We should backport what's in `hackage-server` and then migrate to `Cabal-3.2` or `3.4`.

**Comment by andreasabel** [comment]
> We should backport what's in `hackage-server` and then migrate to `Cabal-3.2` or `3.4`.

@phadej, would you have a pointer to what we should port from `hackage-server` to `hackage-cli`?

**Comment by andreasabel** [comment]
Updated in 
- #37.

**Pull request closed without merging**